### ISSUE TITLE
Category: Search - Reverse Image Search

### ISSUE THREAD

**Issue by CKerswell** [photoprism/photoprism#4497]
Often when searching it is possible to find one image of something, and know you have similar photos that you would like to browse, but can't find them.
Alternatively, you remember taking a photos of something previously and would like to find them again.

It would be great to have two new search options based on images:
1) Upload an image and find similar images.
2) Having found an existing image in the gallery, locate all similar images.

Specific use cases I have where this would be useful:
a) Electricity / Gas meter readings, it would be great to be able to take a new photo of the meter, and PhotoPrism show me all the previous photos of the same meter.
b) Progress of a plant growing in a sealed ecosystem.  I know I have many previous photos of it's progress, so being able to find one of those, or take a new one, and use that to locate all of the previous images.

The auto tagging feature is pretty good at enabling textual search of the gallery, but it's often difficult to work out the exact keywords that are generated (example, a Jam Jar often gets tagged as a "Bottle").

Basically, I'd like Google Reverse Image search, but searching my personal photos :-)

**Comment by graciousgrey** [comment]
Thank you very much for your feedback! 

An option to display similar images and to (optionally) stack them is planned for the future: 
- https://github.com/photoprism/photoprism/issues/28
